### PR TITLE
[Buttons] Add basic Floating Button example

### PR DESCRIPTION
**Pull request by romoore**
Includes automatic adjustment of extended/normal layout depending on size
classes.

ext-fab-iphone4s

ext-fab-ipad-resized

Demonstrates #2495

### CODE REVIEW

**Review by ianegordon — DISMISSED** [review]

**Comment by ianegordon** [comment]
Please merge when CI is happy.

File: components/Buttons/examples/FloatingButtonExampleSwiftViewController.swift [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+    miniFloatingButton = MDCFloatingButton(frame: .zero, shape: .mini)
+    defaultFloatingButton = MDCFloatingButton()
+    largeIconFloatingButton = MDCFloatingButton()
+    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    miniFloatingButton = MDCFloatingButton(frame: .zero, shape: .mini)
```
Comment (yarneo) [inline components/Buttons/examples/FloatingButtonExampleSwiftViewController.swift]: lgtm! just small nit: due to initialization being the same for both inits, i would probably go down the route of initializing in the declaration. Usually if it is immutable and doesn't rely on `self` my preference is initializing in declaration (but that's just a preference).

**Review by yarneo — DISMISSED** [review]

File: components/Buttons/examples/FloatingButtonExampleSwiftViewController.swift (hunk 1 above)
Comment (romoore) [reply components/Buttons/examples/FloatingButtonExampleSwiftViewController.swift]: Done

**Comment by ianegordon** [comment]
Thanks for adding the ObjC sample.  LGTM.
Please merge when CI is happy.
Woot!

**Pull request merged**